### ISSUE TITLE
Collisions on PrepareRun step

### ISSUE THREAD

**Issue by strangeman** [ansible-semaphore/semaphore#545]
PrepareRun contains two important steps: runGalaxy (fetching all roles from ansible galaxy) and listPlaybookHosts (getting task hosts to avoid collisions in parallel task run). 

Several tasks in the queue can do PrepareRun simultaneously and this can lead to the following collision:

* We launch two tasks (A and B) in the same project
* Task A fetch roles from the galaxy and starting listPlaybookHosts
* Task B at the same time fetches roles from galaxy too (deleting old directories with roles and creating a new one)
* Task A crashes with playbook syntax error (because cannot find some roles, deleted during the re-creation by task B)

**Comment by strangeman** [comment]
Same thing appearing when task A already running (in the main phase) and task B re-creating roles in the same project by `ansible galaxy install`.

**Comment by twhiston** [comment]
The two approaches to a fix to this that spring to mind are:
1) Put each task **execution** into it's own folder then we have to check the code out each time, this obviously provides the best level of separation however this could be painful if you have a lot of playbooks and dependencies to pull. It also means that we definitely need to clean up after a task execution or hdd space requirements are going to get out of hand.

2) Add  a 'lock' file that we check at the start of task execution, if it is present then another task is running and we should skip the step where we pull dependencies and update repos etc..... This means that every possible exit path from the task runner will need to ensure it cleans up the lock file or updates on the next run will be skipped (though of course there are some mitigation strategies we could use)

**Comment by strangeman** [comment]
@twhiston please, take a look at my solution - https://github.com/ansible-semaphore/semaphore/pull/546
I tested it and cannot reproduce collision.

**Comment by strangeman** [comment]
Closed by https://github.com/ansible-semaphore/semaphore/pull/546
